src/App.tsx

```diff
@@ -1,14 +1,61 @@
-import React from 'react';
+import debounce from 'lodash.debounce';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import { PersonList } from './components/PersonList/PersonList';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isFocusInput, setIsFocusInput] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000), [],
+  );
+
+  const selectPerson = useCallback((person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setAppliedQuery(person.name);
+    setIsFocusInput(false);
+  }, []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setIsFocusInput(false);
+
+    if (selectedPerson && event.target.value === selectedPerson.name) {
+      return;
+    }
```

Comment:
Here and below we have code duplication, probably we can move this check to separate variable and reuse it